packages/core/src/tracing/trace.ts

```diff
@@ -141,6 +141,73 @@ export function startActiveSpan<T>(context: TransactionContext, callback: (span:
   return maybePromiseResult;
 }
 
+/**
+ * Similar to `Sentry.startActiveSpan`. Wraps a function with a transaction/span, but does not finish the span
+ * after the function is done automatically. Use the returned `finish` function to finish the span.
+ *
+ * ```js
+ * Sentry.startActiveSpan({ name: 'foo' }, (span, finish) => {
+ *   // do work here
+ *   res.on('finish', () => {
+ *     finish();
+ *   });
+ * });
+ * ```
+ *
+ * The created span is the active span and will be used as parent by other spans created inside the function
+ * and can be accessed via `Sentry.getSpan()`, as long as the function is executed while the scope is active.
+ *
+ * Note that if you have not enabled tracing extensions via `addTracingExtensions`
+ * or you didn't set `tracesSampleRate`, this function will not generate spans
+ * and the `span` returned from the callback will be undefined.
+ */
+export function startActiveSpanManual<T>(
+  context: TransactionContext,
+  callback: (span: Span | undefined, finish: () => void) => T,
+): T {
+  const ctx = { ...context };
+  // If a name is set and a description is not, set the description to the name.
+  if (ctx.name !== undefined && ctx.description === undefined) {
+    ctx.description = ctx.name;
+  }
+
+  const hub = getCurrentHub();
+  const scope = hub.getScope();
+
+  const parentSpan = scope.getSpan();
+
+  function getChildSpanOrTransaction(): Span | undefined {
+    if (!hasTracingEnabled()) {
+      return undefined;
+    }
+    return parentSpan ? parentSpan.startChild(ctx) : hub.startTransaction(ctx);
+  }
```

Comment:
l: Seems like this function is duplicated in 3 (probably even 4 if we cound startSpan) places. Doesn't strictly have to happen but I'd prefer if we extracted that logic.